src/monattack.cpp

```diff
@@ -434,79 +453,88 @@ void mattack::science(monster *z, int index) // I said SCIENCE again!
             }
         }
     }
+    player *foe = dynamic_cast< player* >( target );
+    auto msg_type = foe == &g->u ? m_bad : m_neutral;
     std::vector<int> valid;// List of available attacks
     int free_index;
+    bool seen = g->u_see( z );
     monster tmp(GetMType("mon_manhack"));
-    if (dist == 1) {
+    if( dist == 1 ) {
         valid.push_back(1);    // Shock
     }
-    if (dist <= 2) {
+    // mutate() doesn't like non-players right now
+    // It will mutate NPCs, but it will say it mutated the player
+    if( foe != &g->u && dist <= 2 ) {
         valid.push_back(2);    // Radiation
     }
-    if (!free.empty()) {
+    if( !free.empty() ) {
         valid.push_back(3); // Manhack
         valid.push_back(4); // Acid pool
     }
     valid.push_back(5); // Flavor text
     switch (valid[rng(0, valid.size() - 1)]) { // What kind of attack?
-    case 1: // Shock the player
-        if( !g->u.uncanny_dodge() && !g->u.has_artifact_with(AEP_RESIST_ELECTRICITY) &&
-            !g->u.has_active_bionic("bio_faraday") && !g->u.worn_with_flag("ELECTRIC_IMMUNE") ) {
-            add_msg(m_bad, _("The %s shocks you!"), z->name().c_str());
-            z->moves -= 150;
-            g->u.hurtall(rng(1, 2));
-            if (one_in(6) && !one_in(30 - g->u.str_cur)) {
-                add_msg(m_bad, _("You're paralyzed!"));
-                g->u.moves -= 300;
-            }
-        }
+    case 1: // Shock the target
+        // Just reuse the taze - it's a bit different (shocks torso vs all), but let's go for consistency here
+        taze( z, target );
         break;
     case 2: // Radioactive beam
-        add_msg(m_bad, _("The %s opens it's mouth and a beam shoots towards you!"),
-                z->name().c_str());
+        if( seen ) {
+            add_msg( msg_type, _("The %s opens it's mouth and a beam shoots towards %s!"),
```

Comment:
Don't let the Slayer happen to you!
https://github.com/CleverRaven/Cataclysm-DDA/pull/10286